### PR TITLE
Add support for 1.21

### PR DESCRIPTION
**Pull request by ashduino101**
This should add fairly stable support for 1.21, and implements bindings to the necessary NMS functions.

### CODE REVIEW

**Review by AhmetOzkaan — APPROVED** [review]

File: src/main/java/eu/decentsoftware/holograms/api/utils/reflect/Version.java [hunk 1]
```diff
@@ -35,6 +35,7 @@ public enum Version {
     v1_20_R2(20, "1.20.2"),
     v1_20_R3(20, "1.20.3", "1.20.4"),
     v1_20_R4(20, "1.20.5", "1.20.6"),
+    v1_21_R0(21, "1.21"),
```
Comment (d0by1) [inline src/main/java/eu/decentsoftware/holograms/api/utils/reflect/Version.java:38]: Why break consistency with `V0`? Just make it `v1_21_R1`.

**Comment by tomasalias** [comment]
@ashduino101 Can you build me .jar file?

**Comment by ashduino101** [comment]
Sure, I'll fix the versioning and build a jar tonight

**Comment by ashduino101** [comment]
[DecentHolograms-2.8.9.zip](https://github.com/user-attachments/files/15908050/DecentHolograms-2.8.9.zip)

Here's a zip of prebuilt the jar for whoever wants it (@tomasalias)

**Comment by resi23** [comment]
[13:47:11] [Server thread/INFO]: ResistantX issued server command: /hologram line add test 1 #ICON:PORKCHOP
[13:47:11] [Server thread/ERROR]: [DecentHolograms] Could not invoke method encode in class net.minecraft.network.codec.StreamCodec
[13:47:11] [Server thread/ERROR]: [DecentHolograms] Failed to create new instance of class net.minecraft.network.protocol.game.ClientboundSetEntityDataPacket with parameter types [Ljava.lang.Class;@6206ce86

**Comment by resi23** [comment]
> [DecentHolograms-2.8.9.zip](https://github.com/user-attachments/files/15908050/DecentHolograms-2.8.9.zip)
> 
> Here's a zip of prebuilt the jar for whoever wants it (@tomasalias)

[13:47:11] [Server thread/INFO]: ResistantX issued server command: /hologram line add test 1 #ICON:PORKCHOP
[13:47:11] [Server thread/ERROR]: [DecentHolograms] Could not invoke method encode in class net.minecraft.network.codec.StreamCodec
[13:47:11] [Server thread/ERROR]: [DecentHolograms] Failed to create new instance of class net.minecraft.network.protocol.game.ClientboundSetEntityDataPacket with parameter types [Ljava.lang.Class;@6206ce86

**Comment by d0by1** [comment]
Hello, thank you for the PR but I will now close this in favor of a more complete implementation: https://github.com/DecentSoftware-eu/DecentHolograms/pull/220

**Pull request closed without merging**